pkg/scheduler/core/generic_scheduler.go

```diff
@@ -446,11 +447,17 @@ func (g *genericScheduler) findNodesThatFit(ctx context.Context, state *framewor
 	var filtered []*v1.Node
 	filteredNodesStatuses := framework.NodeToStatusMap{}
 
+	allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
```

Comment:
it would be nice if we can break this function into two: runLocalFilters and runExtendersFilters, this will allow us to remove this ugly indentation.

Reply:
do you mean the conditional `g.framework.HasFilterPlugins()` or something else?

Reply:
yes, and to a lesser extent the one that has ```len(g.extenders) != 0``` below

Reply:
we can just get rid of the former check altogether. We would normally have Filters and, if we don't, the code still works as expected. Even better, we would make sure to honor "numNodesToFind".

The extenders check is quite easy to split. I can do it after we agree on the other thing.

Reply:
sg

Reply:
Done. I simplified a bit, but I didn't end up getting rid of the short circuit for no Filters.